ht3/hw3.py

```diff
@@ -0,0 +1,48 @@
+# https://www.python.org/dev/peps/pep-0570/#logical-ordering
+# Positional-only parameters also have the (minor) benefit of enforcing some logical order when
+# calling interfaces that make use of them. For example, the range function takes all its
+# parameters positionally and disallows forms like:
+
+# range(stop=5, start=0, step=2)
+# range(stop=5, step=2, start=0)
+# range(step=2, start=0, stop=5)
+# range(step=2, stop=5, start=0)
+
+# at the price of disallowing the use of keyword arguments for the (unique) intended order:
+
+# range(start=0, stop=5, step=2)
+"""
+Write a function that accept any sequence (list, string, tuple) of unique values and then
+it behaves as range function:
+
+
+import string
+
+
+assert = custom_range(string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(string.ascii_lowercase, 'g', 'p') == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+
+"""
+
+
+def custom_range(*args):
+    string = args[0]
+
+    stop = string.index(args[1])
+    start = 0
+    step = 1
+
+    if len(args) > 2:
+        start = stop
+        stop = string.index(args[2])
+
+        if len(args) > 3:
```

Comment:
Please double check if indentation is needed for this if statement.

Reply:
It would work both ways - but the idea was that it makes no sense to check whenever we have 4th argument if there are 2 or fewer arguments provided according to the previous if statement.

Reply:
Please check comment higher. Thanks.